### ISSUE TITLE
Add at-rule-property-requirelist

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#3985]
> What is the problem you're trying to solve?

Error/warning on missing `font-display` property in `@font-face` at-rule

@font-face {
  font-family: 'Arvo';
  font-display: auto;
  src: local('Arvo'), url(https://fonts.gstatic.com/s/arvo/v9/rC7kKhY-eUDY-ucISTIf5PesZW2xOQ-xsNqO47m55DA.woff2) format('woff2');
}

https://developers.google.com/web/updates/2016/02/font-display

> What solution would you like to see?

New rule with name `require-font-enforce` or `font-display-require` with option `true` (warning when missing) and `false` (warning when exists).

**Comment by cahamilton** [comment]
Mind if I take a look at this one?

**Comment by evilebottnawi** [comment]
@cahamilton feel free to send a PR :+1:

**Comment by evilebottnawi** [comment]
I'm not sure about the name, but it is not difficult to rename :+1:

**Comment by jeddy3** [comment]
In hindsight, I think we should have used "require" from the start, but we went with "no missing". If I remember correctly it was so that all rules that accept `true` as an option were `*-no-*`. I guess we might as well stick with the convention. 

- Name: `font-face-no-missing-font-display`
- Primary option: `true`
- Secondary options: none
- Message: `Unexpected missing font-display`
- Description: "Disallow missing font-display property in font faces"
- Section: "Possible errors" -> "Font face"

@cahamilton Your PR would be much appreciated! You might want to use the [font-family-no-missing-generic-family-keyword](https://github.com/stylelint/stylelint/blob/master/lib/rules/font-family-no-missing-generic-family-keyword/index.js) rule and [PR](https://github.com/stylelint/stylelint/pull/2930/files) as a blue print.

**Comment by jeddy3** [comment]
Now that I think about it some more, this feels like one of the best practises that should be enforced at the configuration level [as per the VISION doc](https://github.com/stylelint/stylelint/blob/master/VISION.md#limit-language-features).

There's a [whole bunch of _nested at-rules_](https://developer.mozilla.org/en-US/docs/Web/CSS/At-rule) that you might want to enforce a property within.

- Name: `at-rule-property-requirelist`
- Primary option: `object` e.g. `{ "font-face": ["font-display"] }`
- Secondary options: none
- Message: `Expected property "${property}" for at-rule "${atRule}"`
- Description: "Specify a requirelist of properties within at-rules"
- Section: "Limit language features" -> "At-rule"

You can use [declaration-property-value-whitelist](https://github.com/stylelint/stylelint/tree/master/lib/rules/declaration-property-value-whitelist) as a blue print.

**Comment by cahamilton** [comment]
@jeddy3 OK sweet. Might be able to tweak what I have now slightly, to make it work. Sounds like a much more scalable solution. 

Just to clarify though, when it comes to the nested at-rules, a primary option like the following `{ "keyframes": ["transform"] }` would be validated that the property exists in both blocks?

```css
@keyframes slidein {
  from {
    transform: translateY(-100%);
    opacity: 0;
  }

  to {
    transform: translateY(0);
    opacity: 1;
  }
}
```

**Comment by jeddy3** [comment]
> would be validated that the property exists in both blocks?

Good question.

My gut says that is should check _neither_ of the blocks. That would be the responsibility a rule like `selector-property-requirelist`. 

So, this rule will only check _nested at-rules_, e.g. `@page`, `@font-face` etc, and not _doubled nested ones_ like `@keyframes`.  Put another way, this rule will only check the properties of the immediate children. This will cover the use case of `font-display` within `@font-face`. We can deal with any other use cases as they are requested.
